src/ir_passes/lower/utils.rs

```diff
@@ -0,0 +1,104 @@
+use crate::ir::{self, Ctx};
+use calyx_ir as calyx;
+
+type AttrPair = (calyx::Attribute, u64);
+/// A set of interface ports that are required for all components.
+pub(super) const INTERFACE_PORTS: [(AttrPair, (&str, u64, calyx::Direction));
+    2] = [
+    (
+        (calyx::Attribute::Bool(calyx::BoolAttr::Clk), 1),
+        ("clk", 1, calyx::Direction::Input),
+    ),
+    (
+        (calyx::Attribute::Bool(calyx::BoolAttr::Reset), 1),
+        ("reset", 1, calyx::Direction::Input),
+    ),
+];
+
+impl ir::EventIdx {
+    /// Returns the name of this interface port.
+    pub fn interface_name(self, comp: &ir::Component) -> Option<String> {
+        if !comp.get(self).has_interface {
+            return None;
+        }
+
+        Some(
+            comp.src_info
+                .as_ref()
+                .map(|src| src.interface_ports.get(&self).unwrap().to_string())
+                .unwrap_or_else(|| format!("ev{}", self.get())),
+        )
+    }
+}
+
+impl ir::ExprIdx {
```

Comment:
I'm not a huge fan of adding new `impl` on a type defined in another folder. In the future, we'd want `filament-ir` to be a separate crate which will break this.

Reply:
Should we just define a helper function that takes `ir::ExprIdx` instead then?